### PR TITLE
Adding filters into envoy dump tables

### PR DESCRIPTION
**Pull request by xeviknal**
belongs to https://github.com/kiali/kiali/issues/3346

Adding filters to help explore the envoy dump tables for each pod.

proxy-filtering

### CODE REVIEW

**Review by skondkar — CHANGES_REQUESTED** [review]
Tested on Kiali UI :     1.29.0 (55fa282bdcde6138f98a863b5c0ed5da59dc463e)

Looks good in chrome, but some issue in firefox ( version 70 and 68.8)
When selecting values from 'resources' dropdown, from 'all' to 'clusters', it works, and then from 'clusters' to 'listners', it throws some error in UI:

Could not fetch envoy config routes entries for productpage-v1-6b746f74dc-qmsq5.
Details shows: Unable to listen on any of the requested ports: [{15000 15000}]

Screenshot-PortError2

**Comment by skondkar** [comment]
One more observation:
When we select resources as 'Clusters' and then 'Direction' from the dropdown, the next dropdown shows values 'Direction, inbound, outbound and a hyphen as seen in the screenshot
The values should be inbound and outbound.

Screenshot-DirectionDropdown

**Comment by xeviknal** [comment]
@skondkar both changes approached. thanks

**Review by skondkar — APPROVED** [review]
Thanks @xeviknal 
Verified on Kiali UI :  1.29.0 (1c2bce8d305546e82f2036ee9b0e7e8c9528f31b)
LGTM

File: src/components/Envoy/EnvoyModal.tsx [hunk 1]
```diff
@@ -74,17 +78,33 @@ class EnvoyDetailsModal extends React.Component<EnvoyDetailProps, EnvoyDetailSta
...
   }
 
-  componentDidUpdate() {
-    if (this.state.fetch) {
-      this.fetchContent();
+  componentDidUpdate(_prevProps: EnvoyDetailProps, prevState: EnvoyDetailState) {
+    if(this.state.pod.name !== prevState.pod.name || this.state.resource !== prevState.resource) {
+      if (this.state.fetch) {
+        this.fetchContent();
+      }
```
Comment (aljesusg) [inline src/components/Envoy/EnvoyModal.tsx:107]: what do you think about
```ts
 if(this.state.fetch && (this.state.pod.name !== prevState.pod.name || this.state.resource !== prevState.resource)) {
```

**Review by aljesusg — APPROVED** [review]
onlye one minor comment. no block, LGFM

**Comment by lucasponce** [comment]
I guess I may ask a change in the feature.

Filter is single value:
image

But as it happens with app/workload/service lists, multi-value would help a lot:
image

For example, I'd like to filter by port 8080 or 9090:
image

Otherwise I'd see the filter a little bit limited (I guess it's a minor change in the filter code, component is prepared for this).

**Review by lucasponce — CHANGES_REQUESTED** [review]
A small change in the behaviour of the filter.

**Comment by xeviknal** [comment]
@lucasponce comments approached. The behavior now is like the workloads/apps/services/istio lists. Applies OR for the same filter and AND between different filters.

**Review by lucasponce — APPROVED** [review]

**Pull request merged**